### PR TITLE
Improve docs on proper tail call.

### PR DESCRIPTION
**Pull request by jfbastien**

### CODE REVIEW

**Comment by jfbastien** [comment]
As suggested in #189.

File: FutureFeatures.md [hunk 1]
```diff
@@ -50,13 +50,19 @@ possible to use a non-standard ABI for specialized purposes.
 Some types of control flow (especially irreducible and indirect) cannot be
 expressed with maximum efficiency in WebAssembly without patterned output by the
 relooper and [jump-threading](https://en.wikipedia.org/wiki/Jump_threading)
-optimizations in the engine.
+optimizations in the engine. Targets uses for more expressive control flow are:
```
Comment (MikeHolman) [inline FutureFeatures.md]: nit: "Target"

File: FutureFeatures.md [hunk 2]
```diff
@@ -50,13 +50,19 @@ possible to use a non-standard ABI for specialized purposes.
...
+* Language interpreters, which often use computed-`goto`.
+* Functional language support, where guarantee tail call optimization is
```
Comment (MikeHolman) [inline FutureFeatures.md]: nit: "guaranteed"

File: FutureFeatures.md [hunk 3]
```diff
@@ -50,13 +50,19 @@ possible to use a non-standard ABI for specialized purposes.
...
+  expected for correctness and performance.
 
 Options under consideration:
 * No action, `while` and `switch` combined with jump-threading are enough.
 * Just add `goto` (direct and indirect).
-* Add [signature-restricted Proper Tail Calls](FutureFeatures.md#signature-restricted-proper-tail-calls).
 * Add new control-flow primitives that address common patterns.
+* Add signature-restricted Proper Tail Calls.
+* Proper tail call, expanding upon signature-restricted proper tail calls, and
```
Comment (MikeHolman) [inline FutureFeatures.md]: Maybe change this to "Add proper tail calls, ...". Currently the wording is kind of ambiguous in context of the previous point as to whether this is a distinct new point or it's a further explanation of the previous one. My language parser had to do some backtracking to realize this was not a continuation.

**Comment by MikeHolman** [comment]
lgtm, after addressing a couple nitpicks.

**Comment by jfbastien** [comment]
Thanks for the fixes @MikeHolman, that was pretty badly written!

**Pull request merged**